log/field.go

```diff
@@ -0,0 +1,204 @@
+package log
```

Comment:
why not have reference field type explicitly? lazyLogger ultimately still ends up with primitive type of `interface{}`, but Reference is something we should support as a first class citizen imo.